tag.go

```diff
@@ -88,6 +89,22 @@ func New(name string, opts Options) *Tag {
 		delete(tag.Options, "body")
 	}
 
+	if tag.Options["value"] != nil {
+		val := tag.Options["value"]
+
+		format := tag.Options["time-format"]
```

Comment:
move all of this into the `case` statement. no point in doing all of this work if it's not going to be used. also, i think that we can just call it `format` and not `time-format`.